packages/web/src/components/date/DateRange.js

```diff
@@ -103,16 +103,21 @@ class DateRange extends Component {
 				)
 				&& !isEqual(prevProps.selectedValue, selectedValue)
 			) {
-				this.handleDateChange(
-					selectedValue
-						? {
-							start: this.props.selectedValue[0] || '',
-							end: this.props.selectedValue[1] || '',
-						} // prettier-ignore
-						: null,
-					true,
-					this.props,
-				);
+				const modDate = selectedValue
+					? {
+						start: this.props.selectedValue[0] || '',
+						end: this.props.selectedValue[1] || '',
+					} // prettier-ignore
+					: { start: '', end: '' };
+				if (value === undefined || (value && value.start === '' && value.end === '')) {
```

Comment:
@ShahAnuj2610 That seems wrong to me, why you're putting it under the `selectedValue` condition. Whenever user changes the value it should update. Check the controlled behavior implementation in other components.